fem/fespace.hpp

```diff
@@ -526,10 +528,17 @@ class FiniteElementSpace
        is preserved. */
    void ReorderElementToDofTable();
 
+   /** @brief  Generates partial face_dof table.
+
+       The table is only defined for exterior faces that coincide with a boundary.
+       The routine uses the bdrElem_dof table and the mesh boundary information.*/
+   void GenerateFaceDofsFromBdr();
+
    void BuildDofToArrays();
 
    const Table &GetElementToDofTable() const { return *elem_dof; }
    const Table &GetBdrElementToDofTable() const { return *bdrElem_dof; }
+   const Table &GetFaceToDofTable() const { return *face_dof; }
```

Comment:
I'm not sure that returning `Table &` instead of `Table *` is good here.

I understand the motivation for keeping the interface for `GetFaceToDofTable` the same as e.g. `GetBdrElementToDofTable` and `GetElementToDofTable`, but this will just crash if you try to use it on a non-NURBS space.

I think a good rule of thumb is that a pointer type represents a "possibly NULL" value, and a reference type represents a "never NULL" value.

In the case of `GetElementToDofTable`, the `elem_dof` table is formed in `FiniteElementSpace::Constructor`, so (possibly barring some edge cases?), `elem_dof` should be non-NULL. (Obviously, my same concern applies to `GetBdrElementToDofTable`, but I guess it is too late to change that interface. On the other hand, it doesn't look like this function is used anywhere in the MFEM codebase, examples, or miniapps. Maybe it can be changed?)

I'm not sure if consistency with existing APIs here outweighs safety, but I think it's worth thinking about. Another option would be to perform some error checking (e.g. `MFEM_VERIFY`).

Reply:
I understand you point. I added to function just for "symmetry", I don;t think it is used.
So I think it is correct to change it, or it might be removed as well.